### PR TITLE
Replace more MapperService usages in favour of QueryShardContext

### PR DESCRIPTION
**Pull request by javanna**
This commit replaces most of the leftover direct access to MapperService from SearchContext and FetchContext with accessing QueryShardContext instead, which wraps the MapperService and exposes a subset of its functionality needed when executing the different phases of search.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java [hunk 1]
```diff
@@ -77,13 +76,6 @@ public MapperService mapperService() {
         return searchContext.mapperService();
     }
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java:77]: Can we remove FetchContext#mapperService() as well?

**Review by romseygeek — APPROVED** [review]
LGTM! One question but may need to be left for a follow-up.

File: server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java (hunk 1 above)
Comment (javanna) [reply server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java:77]: not yet, there is some leftover usage around indexAnalyzer that I did not address. Do you have suggestions on how to get rid of that?

Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java:77]: QueryShardContext has a getIndexAnalyzer() method, you could use that instead I think?

Comment (javanna) [reply server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java:77]: yes you are right, it's an easy one to remove, will do that

Comment (javanna) [reply server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java:77]: it appears that I forgot to push that one commit to this PR, I opened #64620

**Pull request merged**